web/modules/custom/dpl_opening_hours/src/Mapping/OpeningHoursMapper.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Drupal\dpl_opening_hours\Mapping;
+
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursGET200Response;
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursGETRequest;
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursGETRequestCategory;
+use Drupal\dpl_opening_hours\Model\OpeningHoursInstance;
+use Drupal\node\NodeStorageInterface;
+use Drupal\taxonomy\TermStorageInterface;
+use Safe\DateTime;
+use Safe\DateTimeImmutable;
+
+/**
+ * Mapper between value objects and OpenAPI request/response objects.
+ */
+class OpeningHoursMapper {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    private NodeStorageInterface $branchStorage,
+    private TermStorageInterface $categoryStorage,
+  ) {}
+
+  /**
+   * Map an OpenAPI request to a value object.
+   */
+  public function fromRequest(DplOpeningHoursGETRequest $request) : OpeningHoursInstance {
+    $branch = $this->branchStorage->load($request->getBranchId());
+    if (!$branch || $branch->bundle() !== "branch") {
+      throw new \InvalidArgumentException("Invalid branch id '{$request->getBranchId()}'");
+    }
+
+    $categoryTitle = $request->getCategory()?->getTitle();
+    if (!$categoryTitle) {
+      throw new \InvalidArgumentException('No category title provided');
+    }
+    $categoryTerms = $this->categoryStorage->loadByProperties([
```

Comment:
Is it possible that several categories might have the same name?
I know you're using a `reset()` below, but could this cause some other issues?

Reply:
We aim to ensure uniqueness using `taxonomy_unique`in [83b9166](https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/890/commits/83b9166f885a34c906cf6f1f1434a2e7477910db).